File: examples/outbound_ivr_call_mode.py
from supafone_labs import Supafone, outbound_call_mode_readiness


sf = Supafone(api_key="sf_live_replace_me")

readiness = outbound_call_mode_readiness(
    {"enabled": True},
    {
        "provider": "telnyx",
        "dtmf": True,
        "state_persistence": True,
        "human_detection": True,
        "observability": True,
    },
)
if not readiness["ready"]:
    raise RuntimeError(f"Outbound IVR adapter is not ready: {readiness['reasons']}")

sf.labs.agents.create_outbound(
    {
        "name": "Benefits verification",
        "outbound_call_mode": {
            "enabled": True,
            "max_duration_seconds": 180,
            "max_keypresses": 12,
        },
    }
)

File: examples/outbound-ivr-call-mode.ts
import Supafone, { outboundCallModeReadiness } from "supafone-labs";

const sf = new Supafone({ apiKey: "sf_live_replace_me" });

const readiness = outboundCallModeReadiness(
  { enabled: true },
  {
    provider: "telnyx",
    dtmf: true,
    statePersistence: true,
    humanDetection: true,
    observability: true,
  },
);
if (!readiness.ready) throw new Error(`Outbound IVR adapter is not ready: ${readiness.reasons}`);

await sf.labs.agents.createOutbound({
  name: "Benefits verification",
  outboundCallMode: {
    enabled: true,
    maxDurationSeconds: 180,
    maxKeypresses: 12,
  },
});
